src/file_handling_impl/down_sampling_thread.cpp

```diff
@@ -97,13 +97,13 @@ void DownSamplingThread::downsampleAllOnBasisData ()
     {
         for (int channel = 0; channel < data_.size(); channel++)
         {
-            max_channel_length = std::max (max_channel_length, basis_data_[channel]->size());
+            max_channel_length = std::max<unsigned int>(max_channel_length, basis_data_[channel]->size());
             sample_rates[downsampling_factor].append (basis_data_[channel]->getSampleRatePerUnit() / downsampling_factor);
 
             QSharedPointer<QVector<float32> > raw_data_vector (new QVector<float32> (static_cast<int>(basis_data_[channel]->size () / downsampling_factor) + 1));
 
             QSharedPointer<DataBlock> downsampled_data (new FixedDataBlock (raw_data_vector, sample_rates[downsampling_factor][channel]));
-            basis_data_[channel]->addDownSampledVersion (downsampled_data, downsampling_factor);
+            // basis_data_[channel]->addDownSampledVersion (downsampled_data, downsampling_factor);
```

Comment:
The DataBlock doesn't have a `addDownSampledVersion` function, so this doesn't compile.